narwhal/primary/src/proposer.rs

```diff
@@ -419,6 +419,14 @@ impl Proposer {
             let enough_digests = self.digests.len() >= self.header_num_of_batches_threshold;
             let max_delay_timed_out = max_delay_timer.is_elapsed();
             let min_delay_timed_out = min_delay_timer.is_elapsed();
+            let cond = (max_delay_timed_out
+                || ((enough_digests || min_delay_timed_out) && advance))
+                && enough_parents;
+
+            debug!(
+                "Proposer loop starts: round={} enough_parents={} enough_digests={} advance={} max_delay_timed_out={} min_delay_timed_out={} cond={}", 
+                self.round, enough_parents, enough_digests, advance, max_delay_timed_out, min_delay_timed_out, cond
+            );
 
             if (max_delay_timed_out || ((enough_digests || min_delay_timed_out) && advance))
```

Comment:
maybe rename `cond` and use here as well?

Reply:
Good point!